### ISSUE TITLE
Data frame delineation is arbitrary and ill-defined

### ISSUE THREAD

**Issue by jasta** [progrium/wssh#8]
stdin is being read such that each time data is available in the buffer (up to currently 4096 bytes according to wssh, and possibly smaller values dependent on the kernel's buffer size), it transmits a new WebSocket data frame.  This would make it impossible for wssh to create data frames larger than 4096 bytes, and possibly worse, for the user of wssh to control exactly how data frames are delineated in general.

Say a particular WebSocket framed protocol permitted a rapid firing of many small initialization type messages.  To use wssh to send these frames it may be tempting to simply execute:

(print_hello_msg; print_request_1; print_request_2) | wssh ...

But wssh would quite likely interpret this as 1 data frame comprising all 3 messages, violating this hypothetical protocol and potentially confusing the server.

Currently the only way around this would be to ensure small timing windows between each write, and of course to never exceed 4096 bytes.

The solution to this problem is tricky, however, given that netcat's analogy no longer holds up since netcat doesn't have to worry about how it breaks things up into individual TCP or UDP packets.  It's the job of the application layer to provide it's own framing layer that handles it, so netcat doesn't have to worry.

Perhaps we could do something like add an option to specify a special data frame delimiter or an option to add our own wssh-specific data framing layer?  Seems kludgey though, hmm...

**Comment by progrium** [comment]
Yeah, I think that's about as good as we can get here. Specify a data frame delimiter or a fixed frame size in the arguments. Thoughts?

**Comment by jasta** [comment]
I say we start with the former, a configurable data frame delimiter option, and see if that alone is sufficient.  I can take a crack at that today while I'm working on some other issues.

Also, I have a separate set of changes currently under development that allows us to visualize frames using wssh -vvv (this change is in support of adding a mode to configure text vs binary transmission).  With this, I was able to prove quite easily that the (print_msg_1; print_msg_2) | wssh ... invocation above does in fact combine them into 1 data frame.  I was also able to confirm that at least currently it is possible to avoid this behaviour with a small delay of 1 second.  Of course that's not a solution, but it's important to note :)

**Comment by progrium** [comment]
That's cool!
